### PR TITLE
Move Deployed Contracts Page to App Router

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on enhancing the dashboard and contract management features by improving layout components, adding new functionalities for contract handling, and refining the UI components for better user experience.

### Detailed summary
- Updated layout in `deploy.tsx` and `layout.tsx` to use flexbox.
- Introduced `getAuthTokenWalletAddress` function for account management.
- Created `DeployedContractsPage` and `DeployedContractsTable` components.
- Enhanced `GetStartedWithContractsDeploy` with wallet connection logic.
- Refactored `DeployedContracts` to accept `contractList` and `isPending` props.
- Added loading states and improved user feedback in contract tables.
- Integrated `ImportModal` and button functionalities for contract management.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
* **#4995** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4995?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by jnsdls — COMMENTED** [review]
conceptually this makes sense but the loading time for the `/contracts` page with this change is too painful, we need to find a way for that page to load first while the data inside loads, we could either move the fetch back to client-side (for now) or we could try partial pre-rendering for this page, either could solve it

**Comment by MananTank** [comment]
@jnsdls Wrapped the contracts that fetches the contracts with suspense - you should see the shell of the page instantly now

File: apps/dashboard/src/app/(dashboard)/dashboard/contracts/deploy/page.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { redirect } from "next/navigation";
+import { getAuthTokenWalletAddress } from "../../../../api/lib/getAuthToken";
+import { DeployedContractsPage } from "../../../../team/[team_slug]/[project_slug]/contracts/_components/DeployedContractsPage";
+
+export default function Page() {
+  const accountAddress = getAuthTokenWalletAddress();
+  if (!accountAddress) {
+    return redirect(`/login?next=${encodeURIComponent("/dashboard")}`);
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/dashboard/contracts/deploy/page.tsx:8]: should be able to redirect back to this page, right? `/dashboard/contracts/deploy`

**Review by jnsdls — APPROVED** [review]

File: apps/dashboard/src/app/(dashboard)/dashboard/contracts/deploy/page.tsx (hunk 1 above)
Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/dashboard/contracts/deploy/page.tsx:8]: updated

**Pull request merged**